### DISCUSSION TITLE
Please export Renderer functions

### DISCUSSION THREAD

**Discussion by nyarly** [yuin/goldmark#414] [Ideas]
When writing an extension to goldmark, it would be really helpful to have the default implementation of Renderer (and perhaps Parser) functions.

For instance, in https://github.com/kovetskiy/mark/blob/master/pkg/mark/renderer/htmlblock.go#L81 we find a verbatim copy of `renderHTMLBlock` from goldmark's renderer.

Apart from sheer utility, there's a hidden source of regressions here, as well. Much better if instead of `return r.goldmarkRenderHTMLBlock(...)` a client extension could `return goldmarkrenderer.RenderHTMLBlock(...)`

**Comment by mrueg** [comment]
@yuin Thanks for your work on this project! Do you have any thoughts on this? This would be really helpful and make it easier to use goldmark.
